superset/initialization/__init__.py

```diff
@@ -482,12 +482,25 @@ def init_app(self) -> None:
         self.configure_wtf()
         self.configure_middlewares()
         self.configure_cache()
+        self.set_db_default_isolation()
 
         with self.superset_app.app_context():
             self.init_app_in_ctx()
 
         self.post_init()
 
+    def set_db_default_isolation(self) -> None:
+        # This block sets the default isolation level for mysql to READ COMMITTED if not
+        # specified in the config. You can set your isolation in the config by using
+        # SQLALCHEMY_ENGINE_OPTIONS
+        eng_options = self.config["SQLALCHEMY_ENGINE_OPTIONS"] or {}
+        isolation_level = eng_options.get("isolation_level")
+        if not isolation_level and self.config["SQLALCHEMY_DATABASE_URI"].startswith(
+            "mysql"
+        ):
```

Comment:
I kinda liked John's cleaner way of extracting the backend name:
```python
backend = make_url(SQLALCHEMY_DATABASE_URI).get_backend_name()
```
Also, weren't we supposed to do `READ COMMITTED` by default for Postgres, too?

Reply:
ok, I addressed comments, lover how pre-commit caught something around `make_url` not being safe to use so I found I can use `make_url_safe`